cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkTransactional.java

```diff
@@ -232,6 +260,17 @@ private TransactionalDatasetContext(Transaction transaction,
       this.completion = asyncCommit ? new CountDownLatch(1) : null;
     }
 
+    void useSyncCommit() {
+      // This shouldn't happen as the executeLong will check for isJobStarted before calling this method
```

Comment:
can there ever be a concurrency issue? That is, if the Spark program performs something async?